File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/remote/RemoteConfigurationSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.remote;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableRemoteConfigurationSettings.class)
public interface RemoteConfigurationSettings {

  Optional<String> host();

  Optional<Integer> port();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/configurations/remote/RemoteConfigurationSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.remote

import com.fasterxml.jackson.annotation.JsonProperty

data class RemoteConfigurationSettings(

    @JsonProperty("Host")
    val host: String? = null,

    @JsonProperty("Port")
    val port: Int? = null
)
